src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -132,14 +133,21 @@ private void Dispose(bool disposing)
                     GC.SuppressFinalize(this);
                     _stream.Dispose();
 
-                    // Eat any exceptions from the read-ahead task.  We don't need to log, as we expect
+                    // Eat any exceptions from the read-ahead task. We don't need to log, as we expect
                     // failures from this task due to closing the connection while a read is in progress.
                     ValueTask<int>? readAheadTask = ConsumeReadAheadTask();
                     if (readAheadTask != null)
                     {
                         IgnoreExceptions(readAheadTask.GetValueOrDefault());
                     }
                 }
+                else
+                {
+                    Debug.Assert(_readAheadTask is not ValueTask<int> t
+                        || t.IsCompletedSuccessfully
+                        || t.AsTask().GetAwaiter().GetResult() >= 0, // This may throw on the finalizer thread
```

Comment:
Won't it potentially also block the finalizer, until the act of disposing the stream causes this task to complete?